File: 001/pseudoinverse.py
import numpy as np
import time

start_time = time.time()

y = [[0],[1],[4],[9]]
X = [[1,0],[1,1],[1,2],[1,3]]

y = np.matmul(np.transpose(X), y)
X_2 = np.matmul(np.transpose(X),X)

y_2 = np.matmul(np.linalg.inv(X_2), y)

final_time = time.time()
print(final_time - start_time)

File: 001/pseudoinverse/pseudoinverse.jl
start_time = time_ns()

y = [0; 1; 4; 9]
X = [1 0; 1 1; 1 2; 1 3]

y_2 = transpose(X)*y
X_2 = transpose(X)*X

y_3 = inv(X_2)*y_2

final_time = time_ns()
println("Julia Time - ", (final_time - start_time)/1000000000)
